### PR TITLE
dpl: Move Grid from pointers to vectors

### PR DESCRIPTION
**Pull request by QuantamHD**
I found a destructor related bug in one of Google's test cases. Rather than debug it I figured I'd just move it to vector, and not worry about it.

### CODE REVIEW

File: src/dpl/src/Grid.cpp [hunk 1]
```diff
@@ -183,30 +183,14 @@ void Opendp::initGrid()
...
-          grid_[h_index][y][x].is_hopeless = true;
+          Pixel& pixel = grid_[h_index][y][x];
+          pixel.is_hopeless = true;
         }
       }
     }
   }
 }
 
-void Opendp::deleteGrid()
```
Comment (maliberty) [inline src/dpl/src/Grid.cpp:193]: This should do "grid_.clear();"

**Review by maliberty — CHANGES_REQUESTED** [review]
I would have opted for boost::multi_array which is more efficient than either solution.

**Comment by QuantamHD** [comment]
> I would have opted for boost::multi_array which is more efficient than either solution.

Seems like a good idea, but this one is about correctness, and not performance.

**Comment by QuantamHD** [comment]
@maliberty Is this good to go?

**Review by maliberty — APPROVED** [review]

**Pull request merged**